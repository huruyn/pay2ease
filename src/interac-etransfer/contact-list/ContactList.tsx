/* eslint-disable no-console */
import { useMsal } from '@azure/msal-react';
import {
  Col, Row, Card, Table, Button, Modal,
} from 'react-bootstrap';
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import Api, { Contact } from '../../api';
import { formatContactName } from '../../Helpers';
import CommonPageContainer from '../../common/CommonPageContainer';

const RemoveContactModal: React.FC<{
  show: boolean;
  selectedContact: Contact;
  onHide: VoidFunction;
  onDeleteConfirm: VoidFunction;
}> = ({
  show,
  selectedContact,
  onHide,
  onDeleteConfirm,
}) => (
  <Modal
    show={show}
    size="lg"
    aria-labelledby="contained-modal-title-vcenter"
    centered
    id="remove-contact-modal"
  >
    <Modal.Body>
      <h4 style={{ marginTop: 20 }}>Delete Contact - Verification</h4>
      <p>
        Are you sure you want to remove the contact below from your contact
        list?
      </p>
      <Row style={{ marginTop: 50, marginBottom: 20 }}>
        <Col>
          <span>Name:</span>
        </Col>
        <Col>
          {formatContactName(
            selectedContact?.firstName,
            selectedContact?.lastName,
          )}
        </Col>
      </Row>
      <hr
        style={{
          height: '2px',
          borderWidth: '0px',
          color: 'gray',
          background: 'gray',
        }}
      />
      <Row style={{ marginTop: 20, marginBottom: 20 }}>
        <Col>
          <span>Email Address:</span>
        </Col>
        <Col>{selectedContact?.email}</Col>
      </Row>
      <Row style={{ marginTop: 50 }}>
        <Col style={{ padding: 0, background: 'white' }}>
          <Button
            onClick={onHide}
            style={{ margin: 0, background: 'white', border: 'none' }}
            variant="light"
            className="d-flex"
          >
            <div
              style={{
                width: 20,
                height: 25,
                border: '1px dotted grey',
                textAlign: 'center',
                marginRight: 10,
              }}
            >
              P
            </div>
            Cancel
          </Button>
        </Col>
        <Col style={{ display: 'flex', justifyContent: 'flex-end' }}>
          <Button variant="outline-danger" onClick={onHide}>
            Close
          </Button>
          <Button
            id="remove-contact-button"
            className="btn zippy-btn"
            style={{ marginLeft: 15 }}
            onClick={onDeleteConfirm}
          >
            Delete
          </Button>
        </Col>
      </Row>
    </Modal.Body>
  </Modal>
);

// As we are using TypeScript - there are no need to use propTypes package.
// eslint-disable-next-line react/prop-types
const ContactList = () => {
  const { instance, accounts } = useMsal();

  const [modalShow, setModalShow] = useState(false);
  const [contacts, setContacts] = useState<Contact[]>([]);
  const [selectedContact, setSelectedContact] = useState<{ email: string }>();

  useEffect(() => {
    new Api(instance, accounts[0])
      .listContacts()
      .then((result) => setContacts(result ?? []))
      .catch((error) => console.log('error', error));
  }, []);

  const onDeletePressed = (contact: any) => () => {
    setModalShow(true);
    setSelectedContact(contact);
  };
  const onDeleteConfirm = () => {
    const filtered = contacts?.filter(
      ({ email }: Contact) => email !== selectedContact?.email,
    );
    setContacts(filtered || []);
    setModalShow(false);
  };

  return (
    <div className="page-content">
      <CommonPageContainer title="Contact List">
        <Row>
          <Col md={9}>
            <h4 className="mt-4">Add, Edit or Delete a Contact</h4>
            <ul className="mt-4">
              <li>
                To edit a contact&apos;s details, select the contact&apos;s nameddd
              </li>
              <li>
                <a href="/" className="text-black">
                  Edit your Interac e-Transfer alert preferences
                </a>
              </li>
              <li>
                <Link to="add/">
                  <Button variant="link" className="text-black">
                    Add New Contact
                  </Button>
                </Link>
              </li>
              <li>
                Note: Contacts names can not be identical. Please verify and edit
                (if necessary)
              </li>
            </ul>
          </Col>
          <Col md={3}>
            <Card className="p-3 bg-white mt-2">
              <Card.Title>You can also:</Card.Title>
              <Card.Body>
                <a
                  href="/"
                  className="text-decoration-none text-black"
                  style={{
                    fontSize: 12,
                    marginTop: 5,
                    marginBottom: 5,
                    display: 'block',
                  }}
                >
                  Register for Interac e-Transfer Autodeposit
                  <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                    {'>'}
                  </span>
                </a>
                <a
                  href="/"
                  className="text-decoration-none text-black"
                  style={{
                    fontSize: 12,
                    marginTop: 5,
                    marginBottom: 5,
                    display: 'block',
                  }}
                >
                  Make an additional loan payment
                  <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                    {'>'}
                  </span>
                </a>
                <a
                  href="/"
                  className="text-decoration-none text-black"
                  style={{
                    fontSize: 12,
                    marginTop: 5,
                    marginBottom: 5,
                    display: 'block',
                  }}
                >
                  Order cheques
                  <span style={{ color: 'rgb(220, 53, 69)', marginLeft: 5 }}>
                    {'>'}
                  </span>
                </a>
              </Card.Body>
            </Card>
          </Col>
          <Col md={12}>
            <Table className="mt-4">
              <thead>
                <tr
                  className="bg-light"
                  style={{ borderTop: '1px solid #c5c5c5' }}
                >
                  <th>Contact Name</th>
                  <th>Notification Language</th>
                  <th>Details</th>
                  <th>Delete</th>
                </tr>
              </thead>
              <tbody style={{ borderTop: 'none' }}>
                {contacts.map((item, index) => (
                  <tr key={item.id}>
                    <td>
                      <Link to={`edit/${item.id}`} state={{ item }}>
                        {formatContactName(item?.firstName, item?.lastName)}
                      </Link>
                    </td>
                    <td>English</td>
                    <td>
                      {item.email && <div>{item.email}</div>}
                      {item.phone && <div>{item.phone}</div>}
                    </td>
                    <td>
                      <Button
                        onClick={onDeletePressed(item)}
                        variant="light"
                        className="text-black d-flex"
                        id={`contact-list-${index}`}
                      >
                        <div
                          style={{
                            width: 20,
                            height: 25,
                            border: '1px dotted grey',
                            textAlign: 'center',
                            marginRight: 10,
                          }}
                          title="print"
                        >
                          P
                        </div>
                        Delete
                      </Button>
                    </td>
                  </tr>
                ))}
                <RemoveContactModal
                  show={modalShow}
                  onHide={() => setModalShow(false)}
                  selectedContact={selectedContact as Contact}
                  onDeleteConfirm={onDeleteConfirm}
                />
              </tbody>
            </Table>
            <div
              className="mt-4 p-3"
              style={{
                borderTop: '1px solid #e5e5e5',
                borderBottom: '1px solid #e5e5e5',
              }}
            >
              <Row className="d-flex justify-content-end">
                <Link to="add/">
                  <Button
                    className="zippy-btn d-flex"
                    style={{ width: 'auto' }}
                  >
                    Add New Contact
                  </Button>
                </Link>
              </Row>
            </div>
            <div className="mt-2">
              <div className="d-flex">
                <div
                  style={{
                    width: 20,
                    height: 25,
                    border: '1px dotted grey',
                    textAlign: 'center',
                    marginRight: 10,
                  }}
                >
                  P
                </div>
                <span style={{ fontWeight: 'bold' }}>Note:</span>
              </div>
              <div className="d-flex align-items-end">
                <span>Your use of Interac e-Transfer is subject to the </span>
                <Row>
                  <Col md={12} className="d-flex align-items-center mt-2">
                    <div
                      style={{
                        width: 20,
                        height: 25,
                        border: '1px dotted grey',
                        textAlign: 'center',
                        marginRight: 10,
                      }}
                    >
                      P
                    </div>
                    <span>
                      Interac e-Transfer Terms and Conditions (PDF, 197KB).
                    </span>
                  </Col>
                </Row>
              </div>
            </div>
          </Col>
        </Row>
      </CommonPageContainer>
    </div>
  );
};

export default ContactList;
