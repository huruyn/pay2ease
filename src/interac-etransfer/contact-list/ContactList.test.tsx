import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ContactList from './ContactList';

const ReactTestRenderer = require('react-test-renderer');

describe('Contact List Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <ContactList />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render contact list', () => {
    const { getByText } = render(
      <BrowserRouter>
        <ContactList />
      </BrowserRouter>,
    );
    expect(getByText('Add, Edit or Delete a Contact')).toBeInTheDocument();
  });
});
