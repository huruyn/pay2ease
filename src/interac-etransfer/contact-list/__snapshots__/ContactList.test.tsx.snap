// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact List Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <h3>
      Contact List
    </h3>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-md-9"
      >
        <h4
          className="mt-4"
        >
          Add, Edit or Delete a Contact
        </h4>
        <ul
          className="mt-4"
        >
          <li>
            To edit a contact's details, select the contact's nameddd
          </li>
          <li>
            <a
              className="text-black"
              href="/"
            >
              Edit your Interac e-Transfer alert preferences
            </a>
          </li>
          <li>
            <a
              href="/add/"
              onClick={[Function]}
            >
              <button
                className="text-black btn btn-link"
                disabled={false}
                type="button"
              >
                Add New Contact
              </button>
            </a>
          </li>
          <li>
            Note: Contacts names can not be identical. Please verify and edit (if necessary)
          </li>
        </ul>
      </div>
      <div
        className="col-md-3"
      >
        <div
          className="p-3 bg-white mt-2 card"
        >
          <div
            className="card-title h5"
          >
            You can also:
          </div>
          <div
            className="card-body"
          >
            <a
              className="text-decoration-none text-black"
              href="/"
              style={
                Object {
                  "display": "block",
                  "fontSize": 12,
                  "marginBottom": 5,
                  "marginTop": 5,
                }
              }
            >
              Register for Interac e-Transfer Autodeposit
              <span
                style={
                  Object {
                    "color": "rgb(220, 53, 69)",
                    "marginLeft": 5,
                  }
                }
              >
                &gt;
              </span>
            </a>
            <a
              className="text-decoration-none text-black"
              href="/"
              style={
                Object {
                  "display": "block",
                  "fontSize": 12,
                  "marginBottom": 5,
                  "marginTop": 5,
                }
              }
            >
              Make an additional loan payment
              <span
                style={
                  Object {
                    "color": "rgb(220, 53, 69)",
                    "marginLeft": 5,
                  }
                }
              >
                &gt;
              </span>
            </a>
            <a
              className="text-decoration-none text-black"
              href="/"
              style={
                Object {
                  "display": "block",
                  "fontSize": 12,
                  "marginBottom": 5,
                  "marginTop": 5,
                }
              }
            >
              Order cheques
              <span
                style={
                  Object {
                    "color": "rgb(220, 53, 69)",
                    "marginLeft": 5,
                  }
                }
              >
                &gt;
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        className="col-md-12"
      >
        <table
          className="mt-4 table"
        >
          <thead>
            <tr
              className="bg-light"
              style={
                Object {
                  "borderTop": "1px solid #c5c5c5",
                }
              }
            >
              <th>
                Contact Name
              </th>
              <th>
                Notification Language
              </th>
              <th>
                Details
              </th>
              <th>
                Delete
              </th>
            </tr>
          </thead>
          <tbody
            style={
              Object {
                "borderTop": "none",
              }
            }
          />
        </table>
        <div
          className="mt-4 p-3"
          style={
            Object {
              "borderBottom": "1px solid #e5e5e5",
              "borderTop": "1px solid #e5e5e5",
            }
          }
        >
          <div
            className="d-flex justify-content-end row"
          >
            <a
              href="/add/"
              onClick={[Function]}
            >
              <button
                className="zippy-btn d-flex btn btn-primary"
                disabled={false}
                style={
                  Object {
                    "width": "auto",
                  }
                }
                type="button"
              >
                Add New Contact
              </button>
            </a>
          </div>
        </div>
        <div
          className="mt-2"
        >
          <div
            className="d-flex"
          >
            <div
              style={
                Object {
                  "border": "1px dotted grey",
                  "height": 25,
                  "marginRight": 10,
                  "textAlign": "center",
                  "width": 20,
                }
              }
            >
              P
            </div>
            <span
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Note:
            </span>
          </div>
          <div
            className="d-flex align-items-end"
          >
            <span>
              Your use of Interac e-Transfer is subject to the 
            </span>
            <div
              className="row"
            >
              <div
                className="d-flex align-items-center mt-2 col-md-12"
              >
                <div
                  style={
                    Object {
                      "border": "1px dotted grey",
                      "height": 25,
                      "marginRight": 10,
                      "textAlign": "center",
                      "width": 20,
                    }
                  }
                >
                  P
                </div>
                <span>
                  Interac e-Transfer Terms and Conditions (PDF, 197KB).
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer-content-wrapper"
  >
    <div
      className="quick-links container"
    >
      <div
        className="row"
      >
        <h3
          className="col-sm-12"
        >
          Quick Links
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <ul>
            <li>
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li>
              Please Review Our Legal Stuff by visiting our
               
              <a
                className="text-decoration-none"
                href="/legal"
                onClick={[Function]}
              >
                Privacy Policy | TOU | EULA | CardHolder T&C Pages
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-sm-2"
        >
          8465 W Sahara Ave. Las Vegas, NEVADA
          <ul>
            <li>
              <a
                href="tel:+17026231876"
              >
                1.702.623.1876
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@zippy.cash"
              >
                contact@zippy.cash
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 col-sm-12 col-xs-12 text-left"
        >
          © 2021 Zippy Cash LLC  All rights reserved.
          <br />
          <br />
          The Program Manager for the Zippy Cash® Prepaid Visa Debit Card is PayQuicker, LLC
          <br />
          Zippy Cash® is a registered trademark of Zippy Cash, LLC
          <br />
          The Zippy Cash Prepaid Visa Debit Card is issued by Metropolitan Commercial Bank (Member FDIC) pursuant to a license from Visa U.S.A. Inc. “Metropolitan Commercial Bank” and “Metropolitan” are registered trademarks of Metropolitan Commercial Bank ©2014. BY USING THIS CARD YOU AGREE TO THE TERMS AND CONDITIONS OF THE CARDHOLDER AGREEMENT AND FEE SCHEDULE, IF ANY. All trademarks and service marks are the property of their respective owners.
          <br />
          In compliance with the USA PATRIOT Act* and other laws, anyone who is a US resident over 18 with a valid Social Security number and a verifiable physical US address (no PO boxes or APO addresses) can apply for an Zippy Cash Card. This information must be provided, along with other information pertinent to receiving approval, when applying for a Zippy Cash Card. Currently, we do not issue the Zippy Cash Card to residents of Puerto Rico, Guam, the Virgin Islands, American Samoa, or other US territories.
          <br />
          *Important information for opening a card account: To help the federal government fight the funding of terrorism and money laundering activities, the USA PATRIOT Act requires all financial institutions and their third parties to obtain, verify and record information that identifies each person who opens a card account. What this means for you: When you open a card account, we will ask for your name, address, date of birth, social security number or country identification number, and other information that will allows us to identify you. We may also request a copy of your driver’s license or other documentation bearing your photo as verification of your identity.
          <br />
          Apple and the Apple logo are trademarks of Apple Inc., registered in the U.S. and other countries. App Store is a service mark of Apple Inc. Google Play is a trademark of Google Inc. Data rates may apply. Samsung Pay is a trademark of Samsung Inc.
        </div>
      </div>
    </div>
    <footer
      className="container mb-3"
    >
      <div
        className="d-none d-md-block"
      >
        <div
          className="d-flex justify-content-between border-bottom"
        />
        <div
          className="d-flex justify-content-between border-top"
        >
          <div
            className=""
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
          <div
            className="copyright mt-1"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
      <div
        className="d-block d-md-none"
      >
        <div
          className="d-flex border-bottom"
        >
          <div
            className="d-flex flex-column me-2"
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
        </div>
        <div
          className="d-flex justify-content-end border-top"
        >
          <div
            className="copyright mt-2 text-end small"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
